File: src/react_app/src/interface/admin/NoticeNews.ts
export type NoticeNewsPostType = 'NEWS' | 'NOTICE';

export interface NoticeNewsListItem {
  noticeNewsSeq: number;
  postType: NoticeNewsPostType;
  title: string;
  postDate: string;
  viewCount: number;
  topYn: string;
  topStartDate?: string | null;
  topEndDate?: string | null;
  useYn: string;
  fileSeq: number | null;
  attachmentCount: number;
  rgstUserName?: string | null;
  rgstDateTime?: string | null;
  uptDateTime?: string | null;
}

export interface NoticeNewsDetail {
  noticeNewsSeq: number;
  postType: NoticeNewsPostType;
  title: string;
  content: string | null;
  postDate: string;
  viewCount: number;
  topYn: string;
  topStartDate?: string | null;
  topEndDate?: string | null;
  useYn: string;
  fileSeq: number | null;
  rgstUserSeq?: number | null;
  rgstUserName?: string | null;
  rgstDateTime?: string | null;
  uptUserSeq?: number | null;
  uptUserName?: string | null;
  uptDateTime?: string | null;
}

export interface NoticeNewsSavePayload {
  postType: NoticeNewsPostType;
  title: string;
  content: string;
  postDate: string;
  topYn: string;
  topStartDate?: string | null;
  topEndDate?: string | null;
  useYn: string;
  fileSeq: number | null;
}

export interface NoticeNewsSearchParams {
  postType?: NoticeNewsPostType | '';
  useYn?: string;
  keyword?: string;
}

export const NOTICE_NEWS_POST_TYPE_LABEL: Record<NoticeNewsPostType, string> = {
  NEWS: 'News',
  NOTICE: 'Notice',
};

export const NOTICE_NEWS_POST_TYPE_TONE: Record<NoticeNewsPostType, 'blue' | 'green'> = {
  NEWS: 'blue',
  NOTICE: 'green',
};

File: src/main/java/com/kcabEvent/domain/NoticeNews.java
package com.kcabEvent.domain;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
public class NoticeNews {
    private Long noticeNewsSeq;
    private String postType;
    private String title;
    private String content;
    private LocalDate postDate;
    private Integer viewCount;
    private String topYn;
    private LocalDate topStartDate;
    private LocalDate topEndDate;
    private String useYn;
    /** 첨부파일 그룹 (event.tb_file.file_seq) */
    private Long fileSeq;
    /** DB: event.notice_news.created_by */
    private Long rgstUserSeq;
    /** DB: event.notice_news.created_at */
    private LocalDateTime rgstDateTime;
    /** DB: event.notice_news.updated_by */
    private Long uptUserSeq;
    /** DB: event.notice_news.updated_at */
    private LocalDateTime uptDateTime;
}
